src/django/pfb_analysis/migrations/0032_reimport_results.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-11-20 15:54
+from __future__ import unicode_literals
+
+import logging
+import os
+import shutil
+import tempfile
+import zipfile
+
+import boto3
+
+from django.conf import settings
+from django.contrib.gis.utils import LayerMapping
+from django.db import migrations
+
+logger = logging.getLogger(__name__)
+
+
+CENSUS_BLOCK_LAYER_MAPPING = {
+    'geom': 'POLYGON',
+    'overall_score': 'OVERALL_SC'
+}
+
+NEIGHBORHOOD_WAYS_LAYER_MAPPING = {
+    'geom': 'LINESTRING',
+    'tf_seg_str': 'TF_SEG_STR',
+    'ft_seg_str': 'FT_SEG_STR',
+    'xwalk': 'XWALK',
+    'ft_bike_in': 'FT_BIKE_IN',
+    'tf_bike_in': 'TF_BIKE_IN',
+    'functional': 'FUNCTIONAL'
+}
+
+
+def geom_from_results_url(shapefile_key):
+    """ Downloads and extracts a zipped shapefile and returns the containing temporary directory.
+    """
+    logger.info('importing results back from shapefile: {sfile}'.format(sfile=shapefile_key))
+    tmpdir = tempfile.mkdtemp()
+    local_zipfile = os.path.join(tmpdir, 'shapefile.zip')
+    s3_client = boto3.client('s3')
+    s3_client.download_file(settings.AWS_STORAGE_BUCKET_NAME,
+                            shapefile_key,
+                            local_zipfile)
+    with zipfile.ZipFile(local_zipfile, 'r') as zip_handle:
+        zip_handle.extractall(tmpdir)
+    return tmpdir
+
+
+def s3_job_url(job, filename):
+    return 'results/{uuid}/{filename}'.format(uuid=job.uuid, filename=filename)
+
+
+def add_results_geoms(apps, schema_editor):
```

Comment:
This core of this data migration is really a reusable piece of code that "imports the geometries for a single analysis job" for each analysis job in the database. Like many other data migrations we've written in the past few months, ideally I'd like to see this migration written as a management command with the core functionality that does the geom import for a single analysis job abstracted to either an AnalysisJob model method (`import_geometries_from_shapefile()`?) or a utility method. 

This provides reusability with very little extra effort (writing the wrapper management command). For instance it would be easier to add support for the job id issue you mentioned in notes without leaving this code orphaned. It would also allow us to easily add management command options for updating individual jobs or a range of jobs (instead of the default of all jobs) in case of failure when we run this code on production.

What do you think?